### PR TITLE
TypeScript migration.

### PR DESCRIPTION
**Pull request by Geljinskiy**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,11 +1,12 @@
...
-
+import React from 'react';
+// local imports
 import { SharedLayout } from 'components';
 import { Home } from 'pages';
-
 import { ROUTES } from 'utils';
 
-function App() {
+const App: React.FC = (): JSX.Element => {
```
Comment (myroslavsuprun) [inline src/App.tsx:9]: Redundant.
Typescript will inherit the types automatically.

File: src/utils/constants/routes.ts [hunk 2]
```diff
@@ -1,4 +1,4 @@
-export const ROUTES = {
+export const ROUTES: Record<string, string> = {
```
Comment (myroslavsuprun) [inline src/utils/constants/routes.ts:1]: Redundant; typescript will inherit the types automatically.

File: src/components/sections/ProductsReview/ProductReviewZX9/ProductReviewZX9.tsx [hunk 3]
```diff
@@ -1,12 +1,18 @@
...
 import { Link } from 'react-router-dom';
-
 import { Button, ProductReviewZX9Image } from 'components';
-
 import { ROUTES } from 'utils';
-
 import css from './ProductReviewZX9.module.scss';
 
-const ProductReviewZX9 = ({ className }) => {
+type ProductReviewZX9Props = {
```
Comment (myroslavsuprun) [inline src/components/sections/ProductsReview/ProductReviewZX9/ProductReviewZX9.tsx:9]: interface IFormSubmitButtonProps {
  validText: string
  invalidText: string
  withoutIcon?: boolean
  notDisabledIfDirty?: boolean
}

const FormSubmitButton = ({
  validText,
  invalidText,
  withoutIcon,
  notDisabledIfDirty,
}: IFormSubmitButtonProps) => {...}

Just simple function argument assigning works as well.

**Review by myroslavsuprun — CHANGES_REQUESTED** [review]

File: src/utils/constants/routes.ts (hunk 2 above)
Comment (Geljinskiy) [reply src/utils/constants/routes.ts:1]: When removing this part of code, I encounter an error.

**Review by myroslavsuprun — APPROVED** [review]

**Pull request merged**